src/GroupManager.php

```diff
@@ -121,13 +137,59 @@ public function removeGroup($entity_type_id, $bundle_id) {
       $editable->set('groups', $groups);
       $saved = $editable->save();
 
+      // Remove all roles associated with this group type.
+      $this->removeRoles($entity_type_id, $bundle_id);
+
       $this->refreshGroupMap();
 
       return $saved;
     }
   }
 
   /**
+   * Creates default roles for the given group type.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID of the group for which to create default roles.
+   * @param string $bundle_id
+   *   The bundle ID of the group for which to create default roles.
+   */
+  protected function createDefaultRoles($entity_type_id, $bundle_id) {
+    $properties = [
+      'group_type' => $entity_type_id,
+      'group_bundle' => $bundle_id,
+    ];
+    foreach ([OgRoleInterface::ANONYMOUS, OgRoleInterface::AUTHENTICATED, OgRoleInterface::ADMINISTRATOR] as $role_name) {
```

Comment:
Here we hardcode `OgRoleInterface::ADMINISTRATOR` - but only `ANONYMOUS` and `AUTHENTICATED` should be. In D7 we have a more flexible system:

```php

// og.api.php

/**
 * Set a default role that will be used as a global role.
 *
 * A global role, is a role that is assigned by default to all new groups.
 */
function hook_og_default_roles() {
  return array('super admin');
}

/**
 * Alter the default roles.
 *
 * The anonymous and authenticated member roles are not alterable.
 *
 * @param $roles
 *   Array with the default roles name.
 */
function hook_og_default_roles_alter(&$roles) {
  // Remove a default role.
  unset($roles['super admin']);
}
```